contracts/test-integration/BlsSigner.test.ts

```diff
@@ -160,6 +160,313 @@ describe("BlsSigner", () => {
+    );
+    expect(transactionBatchResponse.transactions[0].gasLimit).to.equal(
+      BigNumber.from("0x0"),
+    );
+    expect(transactionBatchResponse.transactions[0].value).to.equal(
+      BigNumber.from(expectedAmount),
+    );
+
+    // tx 2
+    expect(transactionBatchResponse.transactions[1])
+      .to.be.an("object")
+      .that.includes({
+        hash: transactionBatchResponse.transactions[1].hash,
+        to: recipients[1],
+        from: blsSigner.wallet.address,
+        data: "0x",
+        chainId: 1337,
+        type: 2,
+        confirmations: 1,
+      });
+    expect(transactionBatchResponse.transactions[1].nonce).to.equal(
+      expectedNonce,
+    );
+    expect(transactionBatchResponse.transactions[1].gasLimit).to.equal(
+      BigNumber.from("0x0"),
+    );
+    expect(transactionBatchResponse.transactions[1].value).to.equal(
+      BigNumber.from(expectedAmount),
+    );
+
+    // tx 3
+    expect(transactionBatchResponse.transactions[2])
+      .to.be.an("object")
+      .that.includes({
+        hash: transactionBatchResponse.transactions[2].hash,
+        to: recipients[2],
+        from: blsSigner.wallet.address,
+        data: "0x",
+        chainId: 1337,
+        type: 2,
+        confirmations: 1,
+      });
+    expect(transactionBatchResponse.transactions[2].nonce).to.equal(
+      expectedNonce,
+    );
+    expect(transactionBatchResponse.transactions[2].gasLimit).to.equal(
+      BigNumber.from("0x0"),
+    );
+    expect(transactionBatchResponse.transactions[2].value).to.equal(
+      BigNumber.from(expectedAmount),
+    );
+  });
+
+  it("should throw an error sending a transaction batch when this.signer is not defined", async () => {
+    // Arrange
+    const newBlsProvider = new Experimental.BlsProvider(
+      aggregatorUrl,
+      verificationGateway,
+      aggregatorUtilities,
+      rpcUrl,
+      network,
+    );
+    const signedTransaction = await blsSigner.signTransactionBatch({
+      transactions: [
+        {
+          value: parseEther("1"),
+          to: ethers.Wallet.createRandom().address,
+        },
+      ],
+    });
+
+    // Act
+    const result = async () =>
+      await newBlsProvider.sendTransactionBatch(signedTransaction);
+
+    // Assert
+    await expect(result()).to.be.rejectedWith(
+      Error,
+      "Call provider.getSigner first",
+    );
+  });
+
+  it("should validate batch options", async () => {
+    // Arrange
+    const batchOptions = {
+      gas: ethers.utils.parseUnits("40000"),
+      maxPriorityFeePerGas: ethers.utils.parseUnits("0.5", "gwei"),
+      maxFeePerGas: ethers.utils.parseUnits("23", "gwei"),
+      nonce: 39,
+      chainId: 1337,
+      accessList: [],
+    };
+
+    // Act
+    const result = await blsSigner._validateBatchOptions(batchOptions);
+
+    // Assert
+    expect(result).to.deep.equal(batchOptions);
+    expect(result.nonce).to.equal(BigNumber.from("39"));
+    expect(result.nonce).to.have.property("add");
```

Comment:
Above assertion can result in false positives, so these 2 extra assertions test "BigNumber.from" worked.
(e.g. before these extra assertions, the test will still pass if "BigNumber.from" is removed in the method which is incorrect behaviour)